packages/zoe/src/contracts/callSpread.js

```diff
@@ -0,0 +1,217 @@
+// @ts-check
+import '../../exported';
+
+import { assert, details } from '@agoric/assert';
+import { makePromiseKit } from '@agoric/promise-kit';
+import { E } from '@agoric/eventual-send';
+import {
+  assertProposalShape,
+  depositToSeat,
+  natSafeMath,
+  trade,
+  assertUsesNatMath,
+} from '../contractSupport';
+
+const { subtract, multiply, floorDivide } = natSafeMath;
+
+/**
+ * Constants for long and short positions.
+ *
+ * @type {{ LONG: 'long', SHORT: 'short' }}
+ */
+const Position = {
+  LONG: 'long',
+  SHORT: 'short',
+};
+
+const PERCENT_BASE = 100;
+const inverse = percent => subtract(PERCENT_BASE, percent);
+
+/**
+ * This contract implements a fully collateralized call spread. This is a
+ * combination of a call option bought at one strike price and a second call
+ * option sold at a higher price. The contracts are sold in pairs, and the
```

Comment:
```suggestion
 * option sold at a higher price. The invitations are sold in pairs, and the
```
I don't think we want to call the invitations contracts.

Reply:
Actually, this contract is not about selling invitations at all - that's a different level. This contract produces invitations.

Reply:
yep. thanks for the correction.